src/react/listen.jsx

```diff
@@ -11,11 +13,19 @@ export default (listeners = defaultListeners) => WrappedComponent => class exten
   }
 
   componentWillMount() {
-    this.context.store.listeners.add(listeners)
+    const { store } = this.context
+
+    store.dispatch(toAction('ADDING'))
+    store.listeners.add(listeners)
+    store.dispatch(toAction('ADDED'))
   }
 
   componentWillUnmount() {
-    this.context.store.listeners.remove(listeners)
+    const { store } = this.context
+
+    store.dispatch(toAction('REMOVING'))
+    store.listeners.remove(listeners)
+    store.dispatch(toAction('REMOVED'))
```

Comment:
And is it really necessary to have `REMOVING` and `REMOVED` ?

Just having only `@@krm/LISTENERS>REMOVED` maybe renamed in `@@krm/LISTENERS>UNMOUNTED`

Reply:
see previous comment